test/Unit/DependencyManagement/DependencyManagementTests.cs

```diff
@@ -177,5 +186,131 @@ public void TestManagedDependencyNoRequirementsFileShouldThrow()
             Assert.Contains("No 'requirements.psd1'", exception.Message);
             Assert.Contains("is found at the FunctionApp root folder", exception.Message);
         }
+
+        [Fact]
+        public void TestManagedDependencySuccessfulModuleDownload()
+        {
+            string filePath = null;
+            try
+            {
+                // Test case setup.
+                var requirementsDirectoryName = "BasicRequirements";
+                var functionFolderPath = Path.Combine(_dependencyManagementDirectory, requirementsDirectoryName, "FunctionDirectory");
+                var functionAppRoot = Path.Combine(_dependencyManagementDirectory, requirementsDirectoryName);
+                var managedDependenciesFolderPath = GetManagedDependenciesPath(functionAppRoot);
+                var functionLoadRequest = GetFuncLoadRequest(functionFolderPath, true);
+
+                // Create DependencyManager and process the requirements.psd1 file at the function app root.
+                var dependencyManager = new TestDependencyManager();
+                dependencyManager.Initialize(functionLoadRequest);
+
+                // Install the function app dependencies.
+                dependencyManager.InstallFunctionAppDependencies(null, _testLogger);
+
+                // Here we will get two logs: one that says that we are installing the dependencies, and one for a successful download.
+                bool correctLogCount = (_testLogger.FullLog.Count == 2);
+                Assert.True(correctLogCount);
+
+                // The first log should say "Installing FunctionApp dependent modules."
+                Assert.Contains(PowerShellWorkerStrings.InstallingFunctionAppDependentModules, _testLogger.FullLog[0]);
+
+                // In the overwritten RunSaveModuleCommand method, we write a text file with module name and version.
+                // Read the file content of the generated file.
+                filePath = Path.Join(DependencyManager.DependenciesPath, dependencyManager.TestFileName);
+                string fileContent = File.ReadAllText(filePath);
+
+                // After running save module, we write a log with the module name and version.
+                // This should match was is written in the log.
+                Assert.Contains(fileContent, _testLogger.FullLog[1]);
+
+                // Lastly, DependencyError should be null since the module was downloaded successfully.
+                Assert.Null(dependencyManager.DependencyError);
+            }
+            finally
+            {
+                TestCaseCleanup();
+                if (filePath != null && File.Exists(filePath))
+                {
+                    try { File.Delete(filePath); } catch { }
+                }
+             }
+        }
+
+        [Fact]
+        public void TestManagedDependencyRetryLogic()
```

Comment:
>TestManagedDependencyRetryLogic [](start = 20, length = 31)

Consider adding a test for the case when the retry logic kicks in and succeeds. In order to do simulate that, I would suggest passing a number to `TestDependencyManager` instead of just a boolean `ShouldThrow` (something like `ShouldThrowCount`).

Reply:
Good point. Thanks for the suggestion. 
I have added a new test case that succeeds after two failed attempts. 

@AnatoliB: Could you please take another look?